addons/test_mail/tests/test_controller_attachment.py

```diff
@@ -1,105 +1,66 @@
-# Part of Odoo. See LICENSE file for full copyright and licensing details.
-from itertools import product
-
 import odoo
 from odoo.addons.mail.tests.common_controllers import MailControllerAttachmentCommon
 
 
 @odoo.tests.tagged("-at_install", "post_install", "mail_controller")
 class TestAttachmentController(MailControllerAttachmentCommon):
     def test_independent_attachment_delete(self):
-        """Test access to delete an attachment"""
-        self._execute_subtests_delete(
-            product(
-                (self.guest, self.user_employee, self.user_portal, self.user_public),
-                (self.WITH_TOKEN, self.NO_TOKEN),
-            ),
-            allowed=False,
-        )
-        self._execute_subtests_delete(
-            product(self.user_admin, (self.WITH_TOKEN, self.NO_TOKEN)),
-            allowed=True,
-        )
-
-    def test_attachment_delete_linked_to_public_thread(self):
-        """Test access to delete an attachment associated with a public thread"""
-        thread = self.env["mail.test.access.public"].create({"name": "Test"})
-        self._execute_subtests_delete(
-            product(
-                (self.guest, self.user_portal, self.user_public),
-                (self.WITH_TOKEN, self.NO_TOKEN),
-            ),
-            allowed=False,
-            thread=thread,
-        )
+        """Test access to delete an attachment whether or not
+        limited `ownership_token` is sent"""
+        # Subtest format: (user, token, result)
         self._execute_subtests_delete(
-            product(
-                (self.user_admin, self.user_employee),
-                (self.WITH_TOKEN, self.NO_TOKEN),
+            (
+                (self.guest, True, True),
+                (self.guest, False, False),
+                (self.user_admin, True, True),
+                (self.user_admin, False, True),
+                (self.user_employee, True, True),
+                (self.user_employee, False, False),
+                (self.user_portal, True, True),
+                (self.user_portal, False, False),
+                (self.user_public, True, True),
+                (self.user_public, False, False),
```

Comment:
same remark for all tests basically, the new format was much better to see common patterns